metrics.md

```diff
@@ -0,0 +1,238 @@
+(
+  uid CHAR(32) NOT NULL encode lzo,    -- a sha256-hashed Firefox Account (FxA) user id
+  s_uid CHAR(32) encode lzo,           -- a surrogate user id (generated by token server)
+  dev CHAR(32) NOT NULL encode lzo,    -- a sha256-hashed FxA device id
+  s_dev CHAR(32) encode lzo,           -- a surrogate device id (generated by token server)
+  ts TIMESTAMP NOT NULL encode lzo,    -- timestamp of the request
+  method VARCHAR(32) encode lzo,       -- request method (GET, POST, etc)
+  code SMALLINT encode lzo,            -- http status code of the response
+  bucket VARCHAR(255) encode bytedict, -- sync bucket name (bookmarks, history, etc)
+  t INTEGER encode bytedict,           -- request time in milliseconds
+  ua_browser VARCHAR(255) encode lzo,  -- request User Agent browser
+  ua_version INTEGER encode lzo,       -- request User Agent browser version
+  ua_os VARCHAR(255) encode lzo,       -- request User Agent OS
+  host VARCHAR(255) encode lzo         -- server hostname that handled the request
+)
+distkey(uid)
+interleaved sortkey(uid, dev);
+```
+
+## Rollups
+
+After a daily table has been loaded, a few sql commands do rollups of the data.
+
+### Device Activity
+
+This table summarizes a day of request activity per device id.
+
+```sql
+create table device_activity
+(
+  uid char(32) not null encode lzo,
+  dev char(32) not null encode lzo,
+  ua_os varchar(8) encode bytedict,
+  ua_browser varchar(16) encode lzo,
+  ua_version integer encode lzo,
+  day date not null encode delta,
+  min_t integer encode bytedict,
+  max_t integer encode lzo,
+  posts integer encode lzo,
+  gets integer encode lzo,
+  puts integer encode lzo,
+  dels integer encode lzo,
+  aoks integer encode lzo,
+  oops integer encode lzo,
+  fups integer encode lzo,
+  r_clients integer encode lzo,
+  r_crypto integer encode lzo,
+  r_forms integer encode lzo,
+  r_history integer encode lzo,
+  r_keys integer encode lzo,
+  r_meta integer encode lzo,
+  r_bookmarks integer encode lzo,
+  r_prefs integer encode lzo,
+  r_tabs integer encode lzo,
+  r_passwords integer encode lzo,
+  r_addons integer encode lzo,
+  w_clients integer encode lzo,
+  w_crypto integer encode lzo,
+  w_forms integer encode lzo,
+  w_history integer encode lzo,
+  w_keys integer encode lzo,
+  w_meta integer encode lzo,
+  w_bookmarks integer encode lzo,
+  w_prefs integer encode lzo,
+  w_tabs integer encode lzo,
+  w_passwords integer encode lzo,
+  w_addons integer encode lzo
+)
+distkey(uid)
+compound sortkey(day, uid);
+```
+
+And the query that inserts today's data.
+
+```sql
+insert into device_activity (
+  select
+    uid,
+    dev,
+    max(ua_os) as ua_os,
+    max(ua_browser) as ua_browser,
+    max(ua_version) as ua_version,
+    trunc(max(ts)) as day,
+    min(t) as min_t,
+    max(t) as max_t,
+    sum(posts) as posts,
+    sum(gets) as gets,
+    sum(puts) as puts,
+    sum(dels) as dels,
+    sum(aoks) as aoks,
+    sum(oops) as oops,
+    sum(fups) as fups,
```

Comment:
I like these names :-P

Where are `fups`, `oops` etc defined?  I don't see them in the subquery.